src/ape/api/query.py

```diff
@@ -94,7 +94,7 @@ def extract_fields(item, columns):
 
 
 class _BaseQuery(BaseModel):
-    columns: List[str]
+    columns: Sequence[str]
```

Comment:
Would this work appropriately as a base model?

Reply:
Yes you can! it is kinda neat. Pydantic will validate it is a sequence but if you give it a list, it will do its normal list validation or if you give it a tuple, it does the normal tuple validation. `mypy` handles the rest in how it gets used and stuff.

Note: we were already passing tuples in here, but they were getting unnecessary converted to lists. Pydantic v2 + mypy told me all about it, even though it is just as much happening now as it is during the refactor (one of those things).

Reply:
https://docs.pydantic.dev/latest/concepts/performance/#sequence-vs-list-or-tuple-mapping-vs-dict

Reply:
You can find examples of Sequence in the v1 docs too like here https://docs.pydantic.dev/1.10/usage/types/#typing-iterables

Reply:
Okay great! Just making sure the research was done here 🫡